### DISCUSSION TITLE
Outline exits with error Command failed with exit code 1

### DISCUSSION THREAD

**Discussion by pradt** [outline/outline#5883] [Self Hosting] [answered]
I'm trying to get this running in docker, using the sample docker compose file that was provided but converted to be used as a stack in portainer. 
However, I'm receiving logs saying that there are two warnings and exits with code 1. I'm not sure where or why its exiting. 

"WRN Retrying redis connection: attempt 1 
WRN The server does not support SSL connections 
error Command failed with exit code 1."

The redis container is stating it is ready "1:M 24 Sep 2023 18:42:01.817 * Server initialized
1:M 24 Sep 2023 18:42:01.817 * Ready to accept connections tcp"

**To Reproduce**
Steps to reproduce the behavior :
Following is the details that I'm using for deployment, I've removed credentials and url information that is identifiable. 

`
version: "3.2"

services:
  outline:
    image: docker.getoutline.com/outlinewiki/outline:latest
    ports:
      - "3000:3000"
    depends_on:
      - postgres
      - redis
      - storage
    environment:
      SECRET_KEY: "" 
      UTILS_SECRET: "" 
      DATABASE_URL: "postgres://:!@postgres:5432/outline"
      REDIS_URL: "redis://redis:6379" 
      URL: ""
      DATA_DIR: '/data'
      #Authentication Services only 1 is required uncomment the variables that you want to use
      #Google Authentication
      #---------------------
      GOOGLE_CLIENT_ID: 
      GOOGLE_CLIENT_SECRET: 
      

  redis:
    image: redis
    ports:
      - "6379:6379"
    volumes:
      - /volume1/docker/outline/redis/redis.conf:/redis.conf
    command: ["redis-server", "/redis.conf"]
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 10s
      timeout: 30s
      retries: 3

  postgres:
    image: postgres
    ports:
      - "5433:5432"
    volumes:
      - /volume1/docker/outline/postgress:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD", "pg_isready"]
      interval: 30s
      timeout: 20s
      retries: 3
    environment:
      POSTGRES_USER: ''
      POSTGRES_PASSWORD: ''
      POSTGRES_DB: 'outline'

  storage:
    image: minio/minio
    ports:
      - "9002:9000"
    entrypoint: sh
    command: -c 'minio server /data'
    volumes:
      - /volume1/docker/outline/storage:/data
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9002/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3

  https-portal:
    image: steveltn/https-portal
    ports:
      - '83:80'
      - '447:443'
    links:
      - outline
      - storage
    restart: always
    volumes:
      - /volume1/docker/outline/https-portal-data:/var/lib/https-portal
    healthcheck:
      test: ["CMD", "service", "nginx", "status"]
      interval: 30s
      timeout: 20s
      retries: 3
    environment:
      DOMAINS: ' -> http://:3000'
      STAGE: 'production'
      WEBSOCKET: 'true'
`

**Expected behavior**
I was expecting it to work. 

**Screenshots**
If applicable, add screenshots or videos to help explain your problem.

**Outline (please complete the following information):**
- Install: self hosted
- Version: latest from dockerhub

**Desktop (please complete the following information):**
 - OS: macOs
 - Browser chrome

**Comment by tommoor** [comment]
Please post the full logs, it's almost always networking issues between containers and nothing to do with Outline

**Comment by dvraja92** [comment, marked as answer]
I was also facing the same issue when setting up locally, and got it working now.

The hint to the real problem is : `WRN The server does not support SSL connections`
So one of the container doesn't support SSL connections, my suspect was "postgres" and when I looked...

There is an option is disable that (If you don't wish to use SSL there), disable it and should be working. 
It should be injected to postgres & outline both env.
```
# Uncomment this to disable SSL for connecting to Postgres
# PGSSLMODE=disable
```


Complete env file : [Env](https://github.com/outline/outline/blob/402695c2e362b95a315db052bb476a5e868d1895/.env.sample#L20C9-L20C9)

**Reply by pradt** [reply]
OMG that did solve it. I was fixed on the Redis being the issue since it said it couldn't connect. Thank you for your assistance with this. @tommoor is it possible to add this as a feature to state why it is erroring out ?

**Reply by tommoor** [reply]
Done – https://github.com/outline/outline/commit/86cb861ca700a81411b6bdad97f867b14c0c1f1e
